libcudacxx/include/cuda/__device/arch_traits.h

```
@@ -472,118 +471,94 @@ template <>
-      return ::cuda::arch::traits<id::sm_100a>();
-    case id::sm_103:
-      return ::cuda::arch::traits<id::sm_103>();
-    case id::sm_103a:
-      return ::cuda::arch::traits<id::sm_103a>();
-    case id::sm_110:
-      return ::cuda::arch::traits<id::sm_110>();
-    case id::sm_110a:
-      return ::cuda::arch::traits<id::sm_110a>();
-    case id::sm_120:
-      return ::cuda::arch::traits<id::sm_120>();
-    case id::sm_120a:
-      return ::cuda::arch::traits<id::sm_120a>();
+    case arch_id::sm_60:
+      return ::cuda::arch_traits<arch_id::sm_60>();
+    case arch_id::sm_61:
+      return ::cuda::arch_traits<arch_id::sm_61>();
+    case arch_id::sm_70:
+      return ::cuda::arch_traits<arch_id::sm_70>();
+    case arch_id::sm_75:
+      return ::cuda::arch_traits<arch_id::sm_75>();
+    case arch_id::sm_80:
+      return ::cuda::arch_traits<arch_id::sm_80>();
+    case arch_id::sm_86:
+      return ::cuda::arch_traits<arch_id::sm_86>();
+    case arch_id::sm_89:
+      return ::cuda::arch_traits<arch_id::sm_89>();
+    case arch_id::sm_90:
+      return ::cuda::arch_traits<arch_id::sm_90>();
+    case arch_id::sm_90a:
+      return ::cuda::arch_traits<arch_id::sm_90a>();
+    case arch_id::sm_100:
+      return ::cuda::arch_traits<arch_id::sm_100>();
+    case arch_id::sm_100a:
+      return ::cuda::arch_traits<arch_id::sm_100a>();
+    case arch_id::sm_103:
+      return ::cuda::arch_traits<arch_id::sm_103>();
+    case arch_id::sm_103a:
+      return ::cuda::arch_traits<arch_id::sm_103a>();
+    case arch_id::sm_110:
+      return ::cuda::arch_traits<arch_id::sm_110>();
+    case arch_id::sm_110a:
+      return ::cuda::arch_traits<arch_id::sm_110a>();
+    case arch_id::sm_120:
+      return ::cuda::arch_traits<arch_id::sm_120>();
+    case arch_id::sm_120a:
+      return ::cuda::arch_traits<arch_id::sm_120a>();
     default:
       ::cuda::__throw_cuda_error(::cudaErrorInvalidValue, "Traits requested for an unknown architecture");
```

Do we want to throw `cuda_error` here? Don't we want to use `cuda::std::invalid_argument` instead?